scripts/check.py

```diff
@@ -1,73 +1,66 @@
 #!/usr/bin/env python3
-import sys
 import os
 import argparse
-from util import run_cmd
+import subprocess
+
+def run_cmd(command):
+    try:
+        output = subprocess.check_output(command, shell=True, stderr=subprocess.STDOUT, text=True)
+        return_code = 0
+    except subprocess.CalledProcessError as e:
+        output = e.output
+        return_code = e.returncode
+    return output, return_code
 
 parser = argparse.ArgumentParser(
                     prog = 'ProgramName',
                     description = 'What the program does',
                     epilog = 'Text at the bottom of help')
 
-def process_args():
-    usage_line = "Usage: python3 check.py <path to build dir> <cpu/igpu/dgpu> <opencl/level0/pocl> <num threads> <num_tries>"
-    if (len(sys.argv) != 6):
-        print("not enough args")
-        print(sys.argv)
-        print(usage_line)
-        sys.exit(1)
+parser.add_argument('work_dir', type=str, help='Path to build directory')
+parser.add_argument('device_type', type=str, choices=['cpu', 'igpu', 'dgpu'], help='Device type')
+parser.add_argument('backend', type=str, choices=['opencl', 'level0-reg', 'level0-imm', 'pocl'], help='Backend to use')
+parser.add_argument('num_threads', type=int, help='Number of threads to use')
+parser.add_argument('num_tries', type=int, help='Number of tries')
```

Comment:
Some of these options could be optional for convenience - like:
* num_threads which defaults to one or number of processor cores in the system.
* num_tries which defaults to one.